compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEControlCodeLowering.cpp

```diff
@@ -23,34 +23,61 @@ struct HalfDmaCpyNdToNpuConverter final
   using OpConversionPattern::OpConversionPattern;
 
   HalfDmaCpyNdToNpuConverter(MLIRContext *context,
-                             const AMDAIE::AMDAIEDeviceModel &deviceModel)
-      : OpConversionPattern(context), deviceModel(std::move(deviceModel)) {
+                             const AMDAIE::AMDAIEDeviceModel &deviceModel,
+                             bool lowerCtrlpktDma)
+      : OpConversionPattern(context),
+        deviceModel(std::move(deviceModel)),
+        lowerCtrlpktDma(lowerCtrlpktDma) {
     minStrideBitWidth = deviceModel.getMinStrideBitWidth();
   }
 
   /// Insert ops to write a BD, patch the address and push it to the queue. This
   /// is specific to Shim BDs for now.
   FailureOr<AMDAIE::NpuPushToQueueOp> insertWriteBdOps(
       AMDAIE::NpuHalfDmaCpyNdOp op, ConversionPatternRewriter &rewriter,
-      AMDAIE::AMDAIETileType tileType,
-      AMDAIE::LogicalObjectFifoFromMemrefOp logicalObjFifo,
-      AMDAIE::BdIdOp bdIdOp, AMDAIE::ChannelOp channelOp, int64_t bufferLength,
-      int64_t bufferOffset, int32_t enablePacket, int32_t packetId,
-      int32_t packetType, SmallVector<OpFoldResult> sizes,
+      AMDAIE::AMDAIETileType tileType, AMDAIE::BdIdOp bdIdOp,
+      AMDAIE::ChannelOp channelOp, int64_t bufferLength, int64_t bufferOffset,
+      int32_t enablePacket, int32_t packetId, int32_t packetType,
+      SmallVector<OpFoldResult> sizes,
       SmallVector<OpFoldResult> strides) const {
     uint8_t numIntraAddrDim = deviceModel.getDmaProp<uint8_t>(
         tileType, AMDAIE::AMDAIEDmaProp::NumAddrDim);
     uint8_t numAddrDim =
         numIntraAddrDim + deviceModel.deviceConfig.dmaNbInterDims;
-    auto subspanOp = dyn_cast_if_present<IREE::HAL::InterfaceBindingSubspanOp>(
-        logicalObjFifo.getMemref().getDefiningOp());
-    if (!subspanOp) {
-      return logicalObjFifo.emitOpError()
-             << "must operate on an `hal.interface.binding.subspan`";
+
+    int64_t argIdx;
+    int64_t elemWidthInBits;
+    uint8_t memSpace;
+    if (lowerCtrlpktDma) {
```

Comment:
You could make these the default values for `argIdx`, `elemWidthInBits` and `memSpace` as they seem reasonable defaults as well (with the comments) or if you prefer it with the if/else maybe add default values for those variables above so they don't have undefined values and potentially cause future non deterministic bugs.